internal/internal_utils.go

```diff
@@ -77,7 +77,12 @@ func (cb *grpcContextBuilder) Build() (context.Context, context.CancelFunc) {
 		ctx = context.Background()
 	}
 	if cb.Headers != nil {
-		ctx = metadata.NewOutgoingContext(ctx, cb.Headers)
+		md, ok := metadata.FromOutgoingContext(ctx)
+		if !ok {
+			ctx = metadata.NewOutgoingContext(ctx, cb.Headers)
+		} else {
+			ctx = metadata.NewOutgoingContext(ctx, metadata.Join(md, cb.Headers))
```

Comment:
Shouldn't we join in reverse order and allow context metadata to overwrite `cb.Headers`?